docs/architecture/tls13-experimental.md

```diff
@@ -66,3 +66,245 @@ together with their level of testing:
+  MBEDTLS_SSL_SERVER_NAME_INDICATION       no
+  MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH       no
+
+  MBEDTLS_ECP_RESTARTABLE                  no
+  MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED     no
+
+  MBEDTLS_KEY_EXCHANGE_PSK_ENABLED         n/a        Make sense in TLS 1.3
+  MBEDTLS_KEY_EXCHANGE_DHE_PSK_ENABLED     n/a        context but their current
+  MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED   n/a        definition is TLS 1.2 only.
+  MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED     n/a
+  MBEDTLS_KEY_EXCHANGE_RSA_ENABLED         n/a
+  MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED     n/a
+  MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED   n/a
+  MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED n/a
+  MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED  n/a
+  MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED    n/a
+  MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED     n/a
+
+  MBEDTLS_USE_PSA_CRYPTO                   no
+
+Not in the plan yet but probably necessary for a viable client:
+- server_name extension
+- support for HelloRetryRequest
+- fallback to TLS 1.2
+
+Coding rules checklist for TLS 1.3
+----------------------------------
+
+The following coding rules are aimed to be a checklist for TLS 1.3 upstreaming
+work to reduce review rounds and the number of comments in each round. They
+come along (do NOT replace) the project coding rules
+(https://tls.mbed.org/kb/development/mbedtls-coding-standards). They have been
+established and discussed following the review of #4882 that was the
+PR upstreaming the first part of TLS 1.3 ClientHello writing code.
+
+TLS 1.3 specific coding rules:
+
+  - TLS 1.3 specific C modules, headers, static functions names are prefixed
+    with `ssl_tls1_3_`. The same applies to structures and types that are
+    internal to C modules.
+
+  - TLS 1.3 specific exported functions, macros, structures and types are
+    prefixed with `mbedtls_ssl_tls1_3_`.
+
+  - The names of macros and variables related to a field or structure in the
+    TLS 1.3 specification should contain as far as possible the field name as
+    it is in the specification. If the field name is `too long` and we prefer
+    to introduce some kind of abbreviation of it, use the same abbreviation
+    everywhere in the code.
+
+    Example 1: #define CLIENT_HELLO_RANDOM_LEN 32, macro for the length of the
+        `random` field of the ClientHello message.
+
+    Example 2 (consistent abbreviation): mbedtls_ssl_tls1_3_write_sig_alg_ext()
+        and MBEDTLS_TLS_EXT_SIG_ALG, `sig_alg` standing for
+        `signature_algorithms`.
+
+  - Regarding vectors that are represented by a length followed by their value
+    in the data exchanged between servers and clients:
+
+    - Use `<vector name>_len` for the name of a variable used to compute the
+      length in bytes of the vector, where <vector name> is the name of the
+      vector as defined in the TLS 1.3 specification.
+
+    - Use `<vector_name>_len_ptr` for the name of a variable intended to hold
```

Comment:
Given this comment and the following one, we would then rather go for:

- <vector_name>_len, length in bytes of the vector.
- p_<vector_name>_len, pointer to the length in bytes of the vector.
- <vector_name>, address of the first byte of the vector value.
- <vector_name>_end, address of the first byte past the vector value.

Did I understand correctly ? Otherwise, I am ok with that.